File: 백준/Silver/2559. 수열/수열.c
#include <stdio.h>
#include <string.h>

int main()
{
	int n, k;

	scanf("%d %d", &n, &k);

	int value[100000];

	for (int i = 0; i < n; i++) {
		scanf("%d", &value[i]);
	}

	int result = 0;
	
	for (int i = 0; i < k; i++) {
	    result += value[i];
	}
	
	int maxValue = result;

	for (int i = k; i < n; i++) {
	    result += value[i] - value[i - k];
		if (maxValue < result) {
			maxValue = result;
		}
	}

	printf("%d", maxValue);

	return 0;
}

File: 백준/Silver/2491. 수열/수열.py
import sys
input = sys.stdin.readline

n = int(input())

list1 = list(map(int, input().split()))

result = 0
result2 = 0

value = list1[0]
value2 = list1[-1]

maxValue = []

for i in range(len(list1)):
    if list1[i] >= value:
        result += 1
        value = list1[i]
        maxValue.append(result)

    else:
        result = 1
        value = list1[i]
        maxValue.append(result)

for s in list1[::-1]:
    if s >= value2:
        result2 += 1
        value2 = s
        maxValue.append(result2)

    else:
        result2 = 1
        value2 = s
        maxValue.append(result2)

print(max(maxValue))